packages/solc-transpiler/src/compiler.ts

```diff
@@ -77,15 +81,22 @@ export const compile = (configJsonString: string, callbacks?: any): string => {
   for (const [filename, fileJson] of Object.entries(res.contracts)) {
     log.debug(`Transpiling file: ${filename}`)
     for (const [contractName, contractJson] of Object.entries(fileJson)) {
-      log.debug(`Transpiling contract: ${contractName}`)
+      // Library links in bytecode strings have invalid hex: they are of the form __$asdfasdf$__.
+      // Because __$ is not a valid hex string, we replace with a valid hex string during transpilation,
+      // storing the links re-substituting the __$* strings afterwards
```

Comment:
Multi-line comments like this should be a method-header in the helper function. The purpose of the helper function is to remove complexity from the flow of this function, but having many lines of reasoning means that the complexity has not been adequately removed.

Reply:
Don't worry about addressing this, though. As mentioned, will clean this up later.